supermarq-benchmarks/supermarq/qcvv/cb.py

```diff
@@ -0,0 +1,800 @@
+    """Process fidelity estimate."""
+
+    _process_fidelity_estimate_std: float | None = None
+    """Standard deviation of the process fidelity estimate."""
+
+    _undressed_process_fidelity_estimate: float | None = None
+    """Undressed process fidelity estimate."""
+
+    _undressed_process_fidelity_estimate_std: float | None = None
+    """Standard deviation of the undressed process fidelity estimate."""
+
+    @property
+    def channel_fidelities(self) -> pd.DataFrame:
+        """Returns the channel fidelities."""
+        if self._channel_fidelities is None:
+            raise self._not_analyzed
+        return self._channel_fidelities
+
+    @property
+    def process_fidelity(self) -> float:
+        """Returns the process fidelity estimate."""
+        if self._process_fidelity_estimate is None:
+            raise self._not_analyzed
+        return self._process_fidelity_estimate
+
+    @property
+    def process_fidelity_std(self) -> float:
+        """Returns the standard deviation of the process fidelity estimate."""
+        if self._process_fidelity_estimate_std is None:
+            raise self._not_analyzed
+        return self._process_fidelity_estimate_std
+
+    @property
+    def undressed_process_fidelity(self) -> float:
+        """Returns the undressed process fidelity estimate.
+
+        Raises:
+            RuntimeError: If the `.analyse()` method has not been ran yet.
+            RuntimeError: If the experiment only concerns the dressed process.
+        """
+        if self.experiment._undressed_process:
+            if self._undressed_process_fidelity_estimate is None:
+                raise self._not_analyzed
+            return self._undressed_process_fidelity_estimate
+        else:
+            raise RuntimeError("Undressed process fidelity is not available for this experiment.")
+
+    @property
+    def undressed_process_fidelity_std(self) -> float:
+        """Returns the undressed process fidelity estimate.
+
+        Raises:
+            RuntimeError: If the `.analyse()` method has not been ran yet.
+            RuntimeError: If the experiment only concerns the dressed process.
+        """
+        if self.experiment._undressed_process:
+            if self._undressed_process_fidelity_estimate_std is None:
+                raise self._not_analyzed
+            return self._undressed_process_fidelity_estimate_std
+        else:
+            raise RuntimeError("Undressed process fidelity is not available for this experiment.")
+
+    def plot_results(
+        self,
+        filename: str | None = None,
+    ) -> plt.Figure:
+        """Plots the experiment data and the corresponding fits.
+
+        Args:
+            filename: An optional file where the plot is saved.
+
+        Returns:
+            A matplotlib figure containing the plots.
+        """
+        processes = self._channel_expectations.process.unique()
+        fig, axs = plt.subplots(len(processes), 2, figsize=(15, 10))
```

Comment:
```suggestion
        fig, axs = plt.subplots(len(processes), 2, figsize=(15, 10), squeeze=False)
```
I think `processes` will always have length > 1, but probably a good idea anyway since we're assuming `axs` is 2D below: https://stackoverflow.com/questions/68202599/weirdness-of-matplotlib-subplot